figures/line.py

```diff
@@ -21,6 +21,10 @@ def _get_line(self, data, key):
             y_axis_label="No. Violations",
         )
         p.line(x, y, legend_label="Progress", line_width=2)
+        try:
+            p.y_range = Range1d(0, (int(max(y)) * 1.1))
+        except ValueError:
+            return
```

Comment:
This will mask a ValueError with a null pointer exception. We should have better error handling here.

Reply:
Any suggestions? I've done this such that no line figure is returned to the main method, and therefore never creates that graph.